staging/vhost-device-sound/src/audio_backends/pipewire/test_utils.rs

```diff
@@ -0,0 +1,129 @@
+// Manos Pitsidianakis <manos.pitsidianakis@linaro.org>
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use std::{
+    path::Path,
+    process::{Child, Command},
+    sync::OnceLock,
+};
+
+use tempfile::{tempdir, TempDir};
+
+/// We want all threads of the same process to share this harness, because cargo
+/// building will block other cargo processes. So build a harness once and share
+/// it with other tests.
+static CELL: OnceLock<PipewireTestHarness> = OnceLock::new();
+
+/// Temporary Dbus session which is killed in drop().
+pub struct DbusSession {
+    pub address: String,
+    pub address_prefix: String,
+}
+
+impl DbusSession {
+    pub fn new(working_dir: &Path) -> Self {
+        let address_prefix = format!("unix:path={}", working_dir.join("dbus").display());
+        let output = Command::new("/usr/bin/dbus-daemon")
+            .args([
+                "--session",
+                "--address",
+                &address_prefix,
+                "--print-address",
+                "--fork",
+            ])
+            .env("DBUS_VERBOSE", "1")
+            .output()
+            .expect("ERROR: dbus-daemon binary not found");
+
+        let mut address = String::from_utf8(output.stdout).unwrap();
+        while address.ends_with('\n') {
+            address.pop();
+        }
+        assert!(
+            address.starts_with(&address_prefix),
+            "Address {} does not start with address prefix `{}`.",
+            address,
+            address_prefix
+        );
+        Self {
+            address,
+            address_prefix,
+        }
+    }
+}
+
+impl Drop for DbusSession {
+    fn drop(&mut self) {
+        println!("INFO: Killing Dbus session {}", self.address);
+        match Command::new("pkill")
```

Comment:
I'm not sure if this is working. IIUC `pkill -f` wants the entire command line, while here IIUC we are passing just `unix:path=...`

Indeed, after 3 runs I see this in the container:
```
root@14ddf9f7cd2e:/workdir/staging# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4624  3456 pts/0    Ss   14:57   0:00 bash
root         622  0.0  0.0   7988  2304 ?        Ss   14:58   0:00 /usr/bin/dbus-dae
root         623  0.0  0.0  36980  7040 pts/0    Sl   14:58   0:00 pipewire
root         669  0.0  0.0   7988  2336 ?        Ss   14:58   0:00 /usr/bin/dbus-dae
root         670  0.0  0.0  36980  7040 pts/0    Sl   14:58   0:00 pipewire
root         713  0.0  0.0   7988  2304 ?        Ss   14:58   0:00 /usr/bin/dbus-dae
root         714  0.0  0.0  36980  7168 pts/0    Sl   14:58   0:00 pipewire
root         727  0.0  0.0   7060  2944 pts/0    R+   14:58   0:00 ps aux
```

Reply:
Would it be better to use PID in some way?

Reply:
Or avoid `--fork` and use `Command::spawn`, so we don't need to call `pkill`